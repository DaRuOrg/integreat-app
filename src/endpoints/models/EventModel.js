export default class EventModel {
  constructor ({ id, title = '', content = '', thumbnail = '', address = '', town = '', date = {} }) {
    this._id = id
    this._title = title
    this._content = content
    this._thumbnail = thumbnail
    this._address = address
    this._town = town
    this._date = date
  }

  get id () {
    return this._id
  }

  get title () {
    return this._title
  }

  get thumbnail () {
    return this._thumbnail
  }

  get content () {
    return this._content
  }

  get address () {
    return this._address + ', ' + this._town
  }

  getDate (locale) {
    return this._date.toLocaleString(locale)
  }
}
