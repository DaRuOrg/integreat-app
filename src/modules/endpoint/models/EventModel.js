// @flow

import moment from 'moment'

type eventModel = {
  id: boolean,
  title: string,
  content: string,
  thumbnail: string,
  availableLanguages: any,
  content: string,
  town: string,
  address: string,
  startDate: moment,
  endDate: moment,
  allDay: boolean,
  excerpt: string
}

class EventModel {
  _id: boolean
  _title: string
  _content: string
  _thumbnail: string
  _availableLanguages: any
  _content: string
  _town: string
  _address: string
  _startDate: moment
  _endDate: moment
  _allDay: boolean
  _excerpt: string

  constructor (eventModel: eventModel) {
    this._id = eventModel.id
    this._title = eventModel.title
    this._content = eventModel.content
    this._thumbnail = eventModel.thumbnail
    this._address = eventModel.address
    this._town = eventModel.town
    this._startDate = eventModel.startDate
    this._endDate = eventModel.endDate
    this._allDay = eventModel.allDay
    this._excerpt = eventModel.excerpt
    this._availableLanguages = eventModel.availableLanguages
  }

  get id (): boolean {
    return this._id
  }

  get title (): string {
    return this._title
  }

  get thumbnail (): string {
    return this._thumbnail
  }

  get content (): string {
    return this._content
  }

  get address (): ?string {
    if (this._address && this._town) {
      return `${this._address}, ${this._town}`
    } else if (this._address) {
      return this._address
    } else if (this._town) {
      return this._town
    }
  }

  get startDate (): moment {
    return this._startDate
  }

  get endDate (): moment {
    return this._endDate
  }

  get allDay (): boolean {
    return this._allDay
  }

  get excerpt (): string {
    return this._excerpt
  }

  get availableLanguages (): any {
    return this._availableLanguages
  }
}

export default EventModel
