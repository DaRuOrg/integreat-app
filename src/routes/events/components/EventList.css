@import url('../../../modules/common/constants/helpers.css');
@import url('../../../modules/common/constants/colors.css');

.list a {
  @mixin remove-a;
}

.first-event {
  border-top: 2px solid var(--theme-color);
}

.event {
  display: flex;
  border-bottom: 2px solid var(--theme-color);
}

.event-thumbnail {
  width: 100px;
  height: 100px;
  flex-shrink: 0;
  padding: 15px 5px;
}

.event-description {
  height: 100%;
  min-width: 1px; /* needed to enable line breaks for too long words, exact value doesn't matter */
  flex-grow: 1;
  margin-left: 15px;
  padding: 15px 0;
  word-wrap: break-word;
}

.event-title {
  font-weight: 900;
}

.event-date {
  padding-bottom: 10px;
  font-weight: 200;
}

.horizontal-line {
  border-bottom: 2px solid var(--theme-color);
}

.no-events {
  display: flex;
  justify-content: center;
  padding-top: 25px;
}
