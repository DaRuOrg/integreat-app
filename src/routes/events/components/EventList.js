import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'

import EventListElement from './EventListElement'
import EventModel from 'modules/endpoint/models/EventModel'
import Caption from 'modules/common/components/Caption'

import style from './EventList.css'

class EventList extends React.Component {
  static propTypes = {
    events: PropTypes.arrayOf(PropTypes.instanceOf(EventModel)).isRequired,
    url: PropTypes.string.isRequired,
    language: PropTypes.string.isRequired
  }

  render () {
    const {t} = this.props
    return (
      <div className={style.list}>
        <Caption title={t('news')}/>
        { this.props.events && this.props.events.length !== 0
          ? this.props.events.map((event, index) =>
            <EventListElement key={event.id}
                   event={event}
                   parentUrl={this.props.url}
                   language={this.props.language}
                   isFirst={index === 0}/>)
          : <div className={style.noEvents}>{t('currentlyNoEvents')}</div>
        }
      </div>
    )
  }
}

export default translate('events')(EventList)
