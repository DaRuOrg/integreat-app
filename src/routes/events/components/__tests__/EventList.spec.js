import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment-timezone'

import EventModel from 'modules/endpoint/models/EventModel'
import EventList from '../EventList'

describe('EventList', () => {
  // we need UTC here, see https://medium.com/front-end-hacking/jest-snapshot-testing-with-dates-and-times-f3badb8f1d87
  // otherwise snapshot testing is not working
  moment.tz.setDefault('UTC')

  const events = [
    new EventModel({
      id: 1234,
      title: 'first Event',
      availableLanguages: {de: '1235', ar: '1236'},
      startDate: moment('2017-11-18 09:30:00'),
      endDate: moment('2017-11-18 19:30:00'),
      allDay: true
    }),
    new EventModel({
      id: 1235,
      title: 'erstes Event',
      availableLanguages: {en: '1234', ar: '1236'},
      startDate: moment('2017-11-18 09:30:00'),
      endDate: moment('2017-11-18 19:30:00'),
      allDay: true
    }),
    new EventModel({
      id: 2,
      title: 'second Event',
      startDate: moment('2017-11-18 09:30:00'),
      endDate: moment('2017-11-18 19:30:00'),
      allDay: true
    })
  ]

  const url = `/augsburg/en/events`

  const language = 'en'

  test('should render a list of events', () => {
    expect(shallow(<EventList events={events} language={language} url={url} />).dive()).toMatchSnapshot()
  })

  test('should render no events', () => {
    expect(shallow(<EventList events={[]} language={language} url={url} />).dive()).toMatchSnapshot()
  })
})
