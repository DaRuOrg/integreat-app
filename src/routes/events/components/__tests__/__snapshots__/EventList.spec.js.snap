// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList should render a list of events 1`] = `
<React.Fragment>
  <Caption
    title="events"
  />
  <div
    className="list"
  >
    <EventListElement
      event={
        EventModel {
          "_address": "",
          "_availableLanguages": Object {
            "ar": "1236",
            "de": "1235",
          },
          "_content": "",
          "_dateModel": DateModel {
            "_allDay": undefined,
            "_endDate": undefined,
            "_startDate": undefined,
          },
          "_excerpt": "",
          "_id": 1234,
          "_thumbnail": "",
          "_title": "first Event",
          "_town": "",
        }
      }
      key="1234"
      language="en"
      parentUrl="/augsburg/en/events"
    />
    <EventListElement
      event={
        EventModel {
          "_address": "",
          "_availableLanguages": Object {
            "ar": "1236",
            "en": "1234",
          },
          "_content": "",
          "_dateModel": DateModel {
            "_allDay": undefined,
            "_endDate": undefined,
            "_startDate": undefined,
          },
          "_excerpt": "",
          "_id": 1235,
          "_thumbnail": "",
          "_title": "erstes Event",
          "_town": "",
        }
      }
      key="1235"
      language="en"
      parentUrl="/augsburg/en/events"
    />
    <EventListElement
      event={
        EventModel {
          "_address": "",
          "_availableLanguages": Array [],
          "_content": "",
          "_dateModel": DateModel {
            "_allDay": undefined,
            "_endDate": undefined,
            "_startDate": undefined,
          },
          "_excerpt": "",
          "_id": 2,
          "_thumbnail": "",
          "_title": "second Event",
          "_town": "",
        }
      }
      key="2"
      language="en"
      parentUrl="/augsburg/en/events"
    />
  </div>
</React.Fragment>
`;

exports[`EventList should render no events 1`] = `
<div>
  <Caption
    title="events"
  />
  <div
    className="noEvents"
  >
    events
  </div>
</div>
`;
